drivers/png/resource_saver_png.cpp

```diff
@@ -53,9 +53,9 @@ Error ResourceSaverPNG::save_image(const String &p_path, const Ref<Image> &p_img
 
 	PoolVector<uint8_t> buffer;
 	Error err = PNGDriverCommon::image_to_png(p_img, buffer);
-	ERR_FAIL_COND_V(err, err);
+	ERR_FAIL_COND_V_MSG(err, err, "Can't convert image to PNG.");
 	FileAccess *file = FileAccess::open(p_path, FileAccess::WRITE, &err);
-	ERR_FAIL_COND_V(err, err);
+	ERR_FAIL_COND_V_MSG(err, err, vformat("Can't save PNG at path: '%s' (missing filename?)", p_path));
```

Comment:
I'm not sure adding "(missing filename?)" is a good idea, that would be just one possible reason for the failure, and it's not very clear to me either. I guess what you mean is that you tried to use it by giving only the path to a folder, and then it failed? In that case I think it's more a documentation issue, maybe the argument should be named "file_path".

For the error itself, I think giving the path is enough for the user to look if the path is valid, and if it is, whether there are permission issues.

Reply:
In my case this wasn't a documentation issue. I was refactoring some old code of mine, and the utility method supposed to append a default filename to the path. So I kinda spend 10 minutes debugging what would be the actual cause for this error.

I like how GDScript gives you these little message hints of what might be the common cause for errors, for instance.

Reply:
Yeah overall I'm not against giving hints, but I prefer that it's done when the hint has a very high percentage of being the correct explanation (> 50%). In this case, permission issues or otherwise invalid path (typoed folder name along the way for example) are probably as likely as misinterpreting the fact that you need to give the path to a file and not a folder :)